src/Sparrow/Json/Parsing/ObjectJsonParser.cs

```diff
@@ -465,6 +493,7 @@ public bool Read()
                 if (current is IDynamicJsonValueConvertible convertible)
                 {
                     current = convertible.ToJson();
+                    continue;
```

Comment:
I added this 'continue' because I think it is missing. @aviv86 - can you verify?